packages/xdl/src/Android.ts

```diff
@@ -263,7 +263,21 @@ export async function installExpoAsync(url?: string) {
 
 export async function uninstallExpoAsync() {
   Logger.global.info('Uninstalling Expo from Android device.');
-  return await getAdbOutputAsync(['uninstall', 'host.exp.exponent']);
+
+  // we need to check if its installed, else we might bump into "Failure [DELETE_FAILED_INTERNAL_ERROR]"
+  const isInstalled = await _isExpoInstalledAsync();
+  if (!isInstalled) {
+    return '';
+  }
+
+  try {
+    return await getAdbOutputAsync(['uninstall', 'host.exp.exponent']);
+  } catch (e) {
+    Logger.global.error(
+      'Could not uninstall Expo from your device, please uninstall Expo manually and try again.'
+    );
+    throw e;
```

Comment:
I put this into the `uninstallExpoAsync` method since this might be used somewhere else. It shows some a generic way that might fix the issue, but that might require some more info. By throwing the error, the execution stops and the error is dumped to the logs.